cmd/ipfs/main.go

```diff
@@ -671,6 +673,15 @@ func apiClientForAddr(addr ma.Multiaddr) (cmdsHttp.Client, error) {
 }
 
 func isConnRefused(err error) bool {
-	return strings.Contains(err.Error(), "connection refused") ||
-		strings.Contains(err.Error(), "target machine actively refused it")
+	urlerr, ok := err.(*url.Error)
+	if !ok {
+		return false
+	}
+
+	netoperr, ok := urlerr.Err.(*net.OpError)
+	if !ok {
+		return false
+	}
+
+	return netoperr.Op == "dial"
```

Comment:
then rename it `isHTTPConnRefused(...)` if you dont want people to use it for other net conn error checking